raspberry_pi_revision.c

```diff
@@ -455,13 +470,59 @@ getRaspberryPiRevision()
 
 //-------------------------------------------------------------------------
 
+int
+getRaspberryPiExtendedRevision()
+{
+    int raspberryPiExtendedRevision = 0;
+
+    FILE *fp = fopen("/proc/device-tree/chosen/rpi-boardrev-ext", "r");
+
+    if (fp == NULL)
+    {
+        perror("/proc/device-tree/chosen/rpi-boardrev-ext");
```

Comment:
I would really prefer to not print errors if the file doesn't exist. It would be good if this still worked on older releases, without error messages.